### PR TITLE
feat(optics): simpler creation

### PR DESCRIPTION
**Pull request by serras**

### CODE REVIEW

**Comment by kutyel** [comment]
Wow this looks really cool! This way we can even simplify more the `path` lens creator!

File: src/operations.js [hunk 1]
```diff
@@ -14,13 +14,27 @@ const combinePreviews = (p1, p2) => (x) => {
...
  * @param {*} optic1
  * @param {*} optic2
  */
 const compose2Optics = (optic1, optic2) => {
-  // start from most specific (iso) to less specific (fold, setter, review)
+  // turn strings and numbers into proper optics
+  if (typeof optic1 == 'string' || optic1 instanceof String) {
+    return compose2Optics(optionalProp(optic1), optic2)
+  }
+  if (typeof optic1 == 'number') {
```
Comment (kutyel) [inline src/operations.js]: Maybe we could also make sure here the value is not `NaN` (stupid JS behaviour) 🤦 
```suggestion
  if (typeof optic1 == 'number' || !isNaN(optic1) {
```
https://flaviocopes.com/how-to-check-value-is-number-javascript/

File: src/operations.js [hunk 2]
```diff
@@ -14,13 +14,27 @@ const combinePreviews = (p1, p2) => (x) => {
...
+    return compose2Optics(optionalIndex(optic1), optic2)
+  }
+  if (typeof optic2 == 'string' || optic2 instanceof String) {
+    return compose2Optics(optic1, optionalProp(optic2))
+  }
+  if (typeof optic2 == 'number') {
```
Comment (kutyel) [inline src/operations.js]: Same as in L27: https://github.com/kutyel/optics.js/pull/7/files#r489566648

File: src/operations.js (hunk 1 above)
Comment (serras) [reply src/operations.js]: Done in the latest commits

**Review by kutyel — APPROVED** [review]

**Pull request merged**